src/Adapter/Form/ChoiceProvider/OrderReturnStateChoiceProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Form\ChoiceProvider;
+
+use OrderReturnState;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturnState\Exception\OrderReturnStateException;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+
+class OrderReturnStateChoiceProvider implements FormChoiceProviderInterface
+{
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @param int $contextLangId
+     */
+    public function __construct(int $contextLangId)
+    {
+        $this->contextLangId = $contextLangId;
+    }
+
+    /**
+     * Get available order return states.
+     *
+     * @return array
+     *
+     * @throws OrderReturnStateException
+     */
+    public function getChoices(): array
+    {
+        $choices = [];
+        $orderStates = OrderReturnState::getOrderReturnStates($this->contextLangId);
+
+        foreach ($orderStates as $orderState) {
+            $choices[sprintf('%s - %s', $orderState['id_order_return_state'], $orderState['name'])]
+                = (int) $orderState['id_order_return_state'];
```

Comment:
Maybe it is worth introducing a var for this case, it is hard to read

```suggestion
            $indexName = sprintf('%s - %s', $orderState['id_order_return_state'], $orderState['name']);
            $choices[$indexName] = (int) $orderState['id_order_return_state'];
```

Reply:
+, hard to understand what you want to do here, if you create var with name that explains what it is then it would take less time to understand what you want to do here.